File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int i=0; i<nums.length; i++){
            set.add(nums[i]);
        }
        int maxCount = 0;
        for(int num : set){
            if(!set.contains(num-1)){
                int newList = 1;
                while(set.contains(num + newList)){
                    newList++;
                }
                maxCount = Math.max(maxCount, newList);
            }
        }
        return maxCount;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-2.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res = nums[0]
        curMax = 1
        curMin = 1

        for num in nums:
            p_max = curMax * num
            curMax = max(num, p_max, num * curMin)
            curMin = min(num, p_max, num * curMin)
            res = max(res, curMax)
        return res